platform/mellanox/mlnx-platform-api/sonic_platform/modules_mgmt.py

```diff
@@ -458,6 +471,38 @@ def power_on_module(self, port, module_sm_obj, dynamic=False):
                 return STATE_HW_NOT_PRESENT
         return STATE_NOT_POWERED
 
+    def is_cmis_api(self, xcvr_api):
+        return isinstance(xcvr_api, cmis.CmisApi)
+
+    def is_sff_api(self, xcvr_api):
+        return isinstance(xcvr_api, sff8636.Sff8636Api) or isinstance(xcvr_api, sff8436.Sff8436Api)
+
+    def is_supported_for_software_control(self, xcvr_api):
+        return self.is_cmis_api(xcvr_api) or self.is_sff_api(xcvr_api)
+
+    def update_frequency(self, port, xcvr_api):
+        # first read the frequency support - if it's 1 then continue, if it's 0 no need to do anything
+        module_fd_freq_support_path = SYSFS_INDEPENDENT_FD_FREQ_SUPPORT.format(port)
+        val_int = utils.read_int_from_file(module_fd_freq_support_path)
+        if 1 == val_int:
+            if isinstance(xcvr_api, cmis.CmisApi):
+                # for CMIS modules, read the module maximum supported clock of Management Comm Interface (MCI) from module EEPROM.
+                # from byte 2 bits 3-2:
+                # 00b means module supports up to 400KHz
+                # 01b means module supports up to 1MHz
+                logger.log_debug(f"check_module_type reading mci max frequency for port {port}")
+                read_mci = xcvr_api.xcvr_eeprom.read_raw(CMIS_MCI_EEPROM_OFFSET, 1)
+                logger.log_debug(f"check_module_type read mci max frequency {read_mci} for port {port}")
+                frequency = read_mci & CMIS_MCI_MASK
```

Comment:
Will it generate 0 or 1? I understand this is existing code, should we open a bug to Doron or fix it by ourself here?

Reply:
I actually tried to run this on some python interpreter online:

CMIS_MCI_MASK = 0b00001100
read_mci = 0b10000000
frequency = read_mci & CMIS_MCI_MASK
print(frequency)

and looks like it does generate 0 or 1

Reply:
```
>>> read_mci = 0b10001000 # read mci is 1MHz
>>> frequency = read_mci & CMIS_MCI_MASK
>>> frequency
8
```